packages/jest-snapshot/src/utils.ts

```diff
@@ -220,15 +220,18 @@ export const saveSnapshotFile = (
   );
 };
 
+const isAnyOrAnything = (input: any) =>
+  ['Any', 'Anything'].includes(input.constructor.name);
```

Comment:
Just wondering, perhaps this check could be more detailed?

```suggestion
const isAnyOrAnything = (input: object) =>
  '$$typeof' in input &&
  input.$$typeof === Symbol.for('jest.asymmetricMatcher') &&
  ['Any', 'Anything'].includes(input.constructor.name);
```

Reference: https://github.com/facebook/jest/blob/470d72ee74e50291bf8593954f24eb75b5621351/packages/expect/src/asymmetricMatchers.ts#L70